src/main/java/io/reactivesocket/ReactiveSocketImpl.java

```diff
@@ -0,0 +1,477 @@
+/**
+ * Copyright 2015 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.reactivesocket;
+
+import io.reactivesocket.internal.Requester;
+import io.reactivesocket.internal.Responder;
+import io.reactivesocket.internal.rx.CompositeCompletable;
+import io.reactivesocket.internal.rx.CompositeDisposable;
+import io.reactivesocket.rx.Completable;
+import io.reactivesocket.rx.Disposable;
+import io.reactivesocket.rx.Observable;
+import io.reactivesocket.rx.Observer;
+import org.agrona.BitUtil;
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Consumer;
+
+import static io.reactivesocket.LeaseGovernor.NULL_LEASE_GOVERNOR;
+
+/**
+ * An implementation of {@link ReactiveSocket}
+ */
+public class ReactiveSocketImpl implements ReactiveSocket {
```

Comment:
I like Netty's convention of naming default implementation as `Default*`, in that case, `DefaultReactiveSocket`.